analyzers/src/SonarAnalyzer.CSharp/Rules/DoNotDecreaseMemberVisibility.cs

```diff
@@ -41,14 +41,16 @@ public sealed class DoNotDecreaseMemberVisibility : SonarDiagnosticAnalyzer
                     }
 
                     var issueFinder = new IssueFinder(classSymbol, c.SemanticModel);
-                    foreach (var diagnostic in classDeclaration.Members.Select(issueFinder.FindIssue).WhereNotNull())
+                    foreach (var issue in classDeclaration.Members.Select(issueFinder.FindIssue).WhereNotNull())
                     {
-                        c.ReportIssue(diagnostic);
+                        c.ReportIssue(Rule, issue.Location, issue.Symbol.ToDisplayString());
                     }
                 },
                 SyntaxKind.ClassDeclaration,
                 SyntaxKindEx.RecordClassDeclaration);
 
+        private record struct Issue(Location Location, ISymbol Symbol);
```

Comment:
Similar to the previous: Change `IssueFinder` to `IssueReporter`, send him the context and let him report the issue directly.